sdk/python/src/dagger/client/_guards.py

```diff
@@ -8,42 +8,32 @@
 from beartype import beartype
 from beartype.door import TypeHint
 from beartype.roar import BeartypeCallHintViolation
-from beartype.vale import IsInstance, IsSubclass
+from beartype.vale import Is, IsInstance, IsSubclass
 
 from .base import Scalar, Type
 
-
-@typing.runtime_checkable
-class FromIDType(typing.Protocol):
-    @classmethod
-    def _id_type(cls) -> Scalar:
-        ...
-
-    @classmethod
-    def _from_id_query_field(cls) -> str:
-        ...
-
-
-IDTypeSubclass = Annotated[FromIDType, IsSubclass[Type, FromIDType]]
-IDTypeSubclassHint = TypeHint(IDTypeSubclass)
-
-
-def is_id_type_subclass(v: type) -> TypeGuard[type[IDTypeSubclass]]:
-    return IDTypeSubclassHint.is_bearable(v)
+IDScalar = Annotated[Scalar, Is[lambda obj: type(obj).__name__.endswith("ID")]]
```

Comment:
(not blocker, just thinking out loud about general issues) Do we need to ban custom objects that end in `ID`? If someone did that in a module today and then wrote a python module that depended on it would it break anything here in particular (besides the breakage server-side, which may make this moot)

Reply:
No, it's not an issue here for two reasons. 🙂

First, this is matching only scalars that end in `ID` and modules today don't create scalars.

Secondly, this is only used in the `HasID` Protocol below and the runtime checking on protocols is pretty basic, it only checks for the existence of the attribute (in this case it should be checking if the object has an `id` method), it doesn't check the types.

So there isn't much value in `IDScalar` here, just static type checking and documentation or just putting it down for later use as the `Is[]` part isn't standard, it's part of the beartype library and I wanted to remember that it supports matching anything with a lambda like this.

Reply:
To be thorough, even if we allow a user to create custom scalars in the near future I don't think they will be able to create a method called `id`, which would have to return a custom scalar ending in `ID`. But even if they could, the worst that could happen would be:
- On serialization that `id` method would be called before sending the result to the server;
- On deserialization it would call `loadXFromID` to return the object, which is added by the server automatically on id-able objects;

Even then there's a chance that the local machinery from `object_type` will kick in for the above before trying to use `id`. Not sure, would have to test for that order but it's something that can be controlled if needed.